### PR TITLE
Tch dashboard

### PR DESCRIPTION
**Pull request by AliSchlereth**
-Added teacher dashboard
--currently basic styled
--students are listed out, but there is not update or create functionality

Question:
I'm thinking this might be a good place to build an internal api and access through ajax. Do you agree? If so is it a bad plan to continue building it out with the rails backend in and come back to this api once other more essential functionality is complete?

### CODE REVIEW

**Comment by AliSchlereth** [comment]
Add DB Seed file
-pass_img (9)
-sentence (60)

**Comment by laurenfazah** [comment]
> I'm thinking this might be a good place to build an internal api and access through ajax. Do you agree? If so is it a bad plan to continue building it out with the rails backend in and come back to this api once other more essential functionality is complete?

Could you give me more detail on what going in either direction would mean? Would you be working on the same functionality either way or are they distinctly different features?

**Comment by AliSchlereth** [comment]
The teacher dashboard will have full CRUD functionality for students. The two options I'm looking at are building it in a students controller and serving it up in a rails view or building it as a students api and then serving it up through an ajax call and some js. 

Since it would be adding more complexity to the original project, I was considering building it in the students controller and once the full mvp is up then going back to refactor it to an api/ajax.  But, that seems like making double work for myself even if it feels like the safer route.  

Does that make any more sense? @laurenfazah

**Comment by laurenfazah** [comment]
Yup, makes sense. I'll leave the decision up to you based on your learning goals. If you'd like to work on JS and AJAX, this is a great opportunity. If you'd rather finish up this feature sooner than later to get working on another feature you'd rather focus on, then creating this as a standard controller with standard rails views is fine - tweaking them down the road shouldn't be too much of a lift (the controller logic would stay essentially the same).

Let me know what you think or if you have more questions/ need help finding direction.

**Comment by laurenfazah** [comment]
I think your best bet is to move that code into an auth/oauth directory of some sort and require it back into app.js.

Also, it looks like you've got some keys exposed on GitHub (unless they're dummy keys). Seeing that in config/auth.js.

**Comment by AliSchlereth** [comment]
I'm not sure where you are looking, I don't have a config/auth.js ? @laurenfazah

**Comment by laurenfazah** [comment]
Hah, sorry I left that second comment by accident. Was meant for someone else. The comment I left right before that still stands though @AliSchlereth

**Comment by AliSchlereth** [comment]
Okay, that's what I thought might be happening.  

I've decided to go ahead and build it as an api and access it with ajax from the get go.  Working on the first (all students by teacher) ajax call now.

**Comment by laurenfazah** [comment]
Sounds great! Let me know if anything comes up that you need help with.

**Comment by AliSchlereth** [comment]
@laurenfazah I have the students by teacher posting all a teachers' students to the tch dashboard. I see it working in the live environment, but my test doesn't pass anymore. When I put in a save and open page the student content isn't there.  Did I do something wrong?   My most current version is pushed up to this branch if you want to check over the code.

**Comment by laurenfazah** [comment]
Which specific test is failing?

**Comment by AliSchlereth** [comment]
Teacher visits their dashboard spec
spec/features/teacher_visits_their_dashboard_spec.rb
@laurenfazah

**Comment by laurenfazah** [comment]
I'm guessing your onGetStudents call isn't resolved by the time Capybara checks for those elements. You can configure Capybara's wait time to be longer than the default - I'd give that a try. If it's working live, I'm guessing this is the issue. https://github.com/jnicklas/capybara

**Comment by AliSchlereth** [comment]
Full CRUD functionality on teacher dashboard for a current_user's (aka Teacher) students.
-build internal api using RESTful routes
-access api through ajax calls
--lists all students upon page load
--create student form
--delete student buttons for each student
--edit student buttons for each student
---edit student button turns student listing into a form
---switches the edit button to an update button
--upon update student section of page is reloaded to display update in correct level positioning

closes #22
